src/app/zap-templates/templates/app/cluster-objects.zapt

```diff
@@ -71,11 +74,42 @@ namespace {{name}} {
 namespace Commands {
 {{/first}}
 namespace {{asUpperCamelCase name}} {
-struct Type {};
+enum FieldId {
+    {{#zcl_command_arguments}}
+    k{{asUpperCamelCase label}}FieldId = {{index}},
+    {{/zcl_command_arguments}}
+};
 
-struct DecodableType {};
-} // namespace {{asUpperCamelCase name}}
+struct Type
+{
+public:
+    static constexpr chip::CommandId sCommandId = {{asUpperCamelCase name}}::Id;
+    //static constexpr chip::ClusterId sClusterId = ::Id;
```

Comment:
@mrjerryjohns @bzbarsky-apple I am not able to get clusterId here, ../../zzz_generated/app-common/app-common/zap-generated/cluster-objects.h:6350:52: error: ‘::Id’ has not been declared
 6350 |     static constexpr chip::ClusterId ClusterId = ::Id;
it seems "CommandName"::Id and "ClusterName"::Id confuse the compiler, in addition, zap zcl_commands don't insert the cluster name inside its context...

In addition, in identification cluster,  there is argument called commandid, so for static constextpr command id, I add prefix s.
    static constexpr chip::CommandId commandId = ExecutionOfACommand::Id;
    // static constexpr chip::ClusterId clusterId = ::Id;
    CommandIdentification commandId;       ^~

Reply:
A few things.  The members we are putting in `lowerCamelCase`, so if we called the member `CommandId` there would be no name collision.  We would have `commandId` and `CommandId`.  If we name it `sCommandId` that _could_ in fact collide with a lowerCamelCase thing at some point....

That said, I do agree that having both `commandId` and `CommandId` there is a bit confusing and it's easy to type the wrong one.  Maybe we should have a `static constexpr GetCommandId()` to make it really different from the struct members, with a comment explaining why?

Reply:
Done, thanks